### ISSUE TITLE
PDAL_DRIVER_PATH required

### ISSUE THREAD

**Issue by jgrocha** [PDAL/PDAL#3832]
**Describe the bug**

I've compiled current master with gcc-11 and run `ctest`. Everything is ok.

After installing pdal, I can not run it without setting `PDAL_DRIVER_PATH`.

```
$ pdal
Usage:
  pdal <options>
  pdal <command> <command options>
  --command        The PDAL command
  --debug          Sets the output level to 3 (option deprecated)
  --verbose, -v    Sets the output level (0-8)
  --drivers        List available drivers
  --help, -h       Display help text
  --list-commands  List available commands
  --version        Show program version
  --options        Show options for specified driver (or 'all')
  --log            Log filename (accepts stderr, stdout, stdlog, devnull as
      special cases)
  --logtiming      Turn on timing for log messages

The following commands are available:
terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
  what():  filesystem error: cannot make canonical path: No such file or directory [./lib]
Aborted (core dumped)
```

Setting `PDAL_DRIVER_PATH` makes pdal happy.

```
jgr@dragon:~$ export PDAL_DRIVER_PATH=/usr/local/lib
jgr@dragon:~$ pdal
Usage:
  pdal <options>
  pdal <command> <command options>
  --command        The PDAL command
  --debug          Sets the output level to 3 (option deprecated)
  --verbose, -v    Sets the output level (0-8)
  --drivers        List available drivers
  --help, -h       Display help text
  --list-commands  List available commands
  --version        Show program version
  --options        Show options for specified driver (or 'all')
  --log            Log filename (accepts stderr, stdout, stdlog, devnull as
      special cases)
  --logtiming      Turn on timing for log messages

The following commands are available:
  - chamfer
  - delta
  - density
  - eval
  - fauxplugin
  - ground
  - hausdorff
  - info
  - merge
  - pipeline
  - random
  - sort
  - split
  - tile
  - tindex
  - translate

See http://pdal.io/apps/ for more detail
```

**Expected**

pdal should run fine without `PDAL_DRIVER_PATH`.

From the [documentation](https://pdal.io/stages/stages.html) (I didn't check the code) it is not clear if the listed paths are only computed based on the current folder or from the binary folder.

I've set `PDAL_DRIVER_PATH` to the `../lib` relative to the binary folder.
```
$ which pdal
/usr/local/bin/pdal
jgr@dragon:~$ echo $PDAL_DRIVER_PATH
/usr/local/lib
```

**Environment**

Ubuntu 22.04, gcc-11(gcc-8 is no longer supported), GDAL 3.6.0dev-f033e1fa1e, released 2022/08/04 (debug build) 

Compiling PDAL

```
cd PDAL/
git pull
rm -rf build; mkdir build; cd build
cmake ..
cmake --build . -j16
sudo cmake --build . --target install
sudo ldconfig
```

**Comment by hobu** [comment]
It doesn't fail this way on OSX. I'll try a docker build next.

**Comment by hobu** [comment]
Fixed in ee902594b30075ebf033fce1e531fc6836965dde. Thanks for the report!
